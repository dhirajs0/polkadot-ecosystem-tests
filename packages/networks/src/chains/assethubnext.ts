import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNext: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNext) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNext = defineChain({
  name: 'assetHubNext',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNext,
  initStorages: getInitStorages(custom.assetHubNext),
})
