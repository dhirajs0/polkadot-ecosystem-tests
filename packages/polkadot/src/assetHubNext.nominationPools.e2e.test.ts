import { assetHubNext } from '@e2e-test/networks/chains'

import { nominationPoolsE2ETests } from '@e2e-test/shared'

nominationPoolsE2ETests(assetHubNext, { testSuiteName: 'AHN Nomination Pools', addressEncoding: 0 })
