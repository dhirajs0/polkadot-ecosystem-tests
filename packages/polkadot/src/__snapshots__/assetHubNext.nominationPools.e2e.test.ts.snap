// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AHN Nomination Pools > nomination pool creation with insufficient funds > create nomination pool with insufficient funds events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x08000000",
          "index": 81,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 4500000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool double join test: an account can only ever be in one pool at a time > join nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 0,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "bonded": 0,
      "joined": true,
      "member": "5HGjWAeFDfFCWPsjFQdVV2Msvz2XtMktvgocEZcCj68kUMaw",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool double join test: an account can only ever be in one pool at a time > join second nomination pool events 1`] = `
[
  {
    "data": {
      "dispatchError": {
        "Module": {
          "error": "0x04000000",
          "index": 81,
        },
      },
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 3600000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool global config test > setting global nomination pool configs with signed origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": 0,
          "refTime": "(rounded 1900000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool lifecycle test > create nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 1000000000,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "depositor": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "method": "Created",
    "section": "nominationPools",
  },
  {
    "data": {
      "bonded": 1000000000,
      "joined": true,
      "member": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool metadata test > create nomination pool events 1`] = `
[
  {
    "data": {
      "amount": 1000000000,
    },
    "method": "Bonded",
    "section": "staking",
  },
  {
    "data": {
      "depositor": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "method": "Created",
    "section": "nominationPools",
  },
  {
    "data": {
      "bonded": 1000000000,
      "joined": true,
      "member": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
    },
    "method": "Bonded",
    "section": "nominationPools",
  },
]
`;

exports[`AHN Nomination Pools > nomination pool metadata test > set metadata events 1`] = `
[
  {
    "data": {
      "caller": "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY",
      "poolId": 1,
    },
    "method": "MetadataUpdated",
    "section": "nominationPools",
  },
]
`;
